roles/translator/src/proxy/bridge.rs

```diff
@@ -244,6 +486,57 @@ impl Bridge {
         })
     }
 
+    async fn handle_new_prev_hash_(
+        self_: Arc<Mutex<Self>>,
+        sv2_set_new_prev_hash: SetNewPrevHash<'static>,
+        tx_status: status::Sender,
+        tx_sv1_notify: broadcast::Sender<server_to_client::Notify<'static>>,
+    ) {
+        loop {
+            let res = self_
+                .safe_lock(|s| s.last_p_hash = Some(sv2_set_new_prev_hash.clone()))
+                .map_err(|_| PoisonLock);
+            handle_result!(tx_status, res);
+            let on_new_prev_hash_res = self_
+                .safe_lock(|s| {
+                    s.channel_factory
+                        .on_new_prev_hash(sv2_set_new_prev_hash.clone())
+                })
+                .map_err(|_| PoisonLock);
+            let on_new_prev_hash_res = handle_result!(tx_status, on_new_prev_hash_res);
+            handle_result!(tx_status, on_new_prev_hash_res);
+
+            let future_jobs = self_
+                .safe_lock(|s| {
+                    let future_jobs = s.future_jobs.clone();
+                    s.future_jobs = vec![];
+                    future_jobs
+                })
+                .map_err(|_| PoisonLock);
+            let mut future_jobs = handle_result!(tx_status, future_jobs);
+
+            while let Some(job) = future_jobs.pop() {
+                if job.job_id == sv2_set_new_prev_hash.job_id {
+                    // Create the mining.notify to be sent to the Downstream.
+                    let notify = crate::proxy::next_mining_notify::create_notify(
+                        sv2_set_new_prev_hash.clone(),
+                        job,
+                    );
+                    // Get the sender to send the mining.notify to the Downstream
+                    handle_result!(tx_status, tx_sv1_notify.send(notify.clone()));
+                    let res = self_
+                        .safe_lock(|s| {
+                            s.last_notify = Some(notify);
+                        })
+                        .map_err(|_| PoisonLock);
+                    handle_result!(tx_status, res);
+                    break;
+                }
```

Comment:
What should we do if there are no future jobs? I feel like we should at least log something.

Reply:
This is a pretty common situation it happens every time that a block is found by another pool for example. Maybe is better a debug here. For now I go with debug, in case I change it to normal log.